### DISCUSSION TITLE
how to visualize and use ReceiveSysex

### DISCUSSION THREAD

**Discussion by tgarello** [bonsai-rx/bonsai#1285] [Q&A]
Hi 

Context:
I want to program an optogenetic experiment. The experiment involve several Arduino and LEDs, and the pulses are in the order of the ms. In the end, it should be a closed loop control system, where the stimuli vary (period and frequency of the pulses).
-> I figured I cannot use Bonsai to turn on/off the LEDs individually because the delay would be too long
-> the stimuli parameters cannot be hard coded into the Arduino has it should vary depending on the image analysis

Problem:
[example.zip](https://github.com/bonsai-rx/bonsai/files/10980924/example.zip)

I am trying to use the firmata protocol to communicate between bonsai and Arduino. I want to implement a few user features (feature IDs : 01H - 0FH). I would use the SendSysex node to communicate the parameters to Arduino. To debug, I uploaded the 
Firmata Arduino example 'EchString.ino'. I know the Arduino receive the Sysex, but I am not sure what the payload contains. So the echo example could help me debug my payload. But I cannot visualize the echoed Sysex in Bonsai.

I tried to use ReceiveSysex node. But I don't know how to visualize the <[Byte][]>.

Is there example how to use ReceiveSysex ?

Thanks a  lot for your help
<img width="168" alt="example" src="https://user-images.githubusercontent.com/120011964/225338807-764f005c-1630-40ac-88a7-d91c7102dac1.png">

Cheers

**Comment by bruno-f-cruz** [comment]
You can either use Merge(reactive) or Index to view individual elements of arrays.
Also, to save it you should use a matrixwriter that will log the raw byte stream to a binary file. 
Hope it helps
